Forge.TreeWalker.UnitTests/test/TreeSchemaValidatorTests.cs

```diff
@@ -0,0 +1,208 @@
+        public void Test_ValidateInvalidSchemaInString()
+        {
+            bool res = ForgeSchemaValidator.ValidateSchemaAsString(tardigradeSchemaString, jschemaRules, false, out IList<ValidationError> errorList);
+            Assert.AreEqual(true, res);
+            Assert.AreEqual(0, errorList.Count);
+        }
+
+        [TestMethod]
+        public void Test_ValidateInvalidSchemaDirectoryPath()
+        {
+            try
+            {
+                ForgeSchemaValidator.ValidateSchemaFromDirectory(invalidSchemaDirectoryPath, jschemaRules, false, out IList<ValidationError> errorList);
+            }
+            catch (Exception e)
+            {
+                Assert.AreEqual("The directory name is invalid.\r\n", e.Message);
+            }
+        }
+
+        [TestMethod]
+        public void Test_ValidateSchemaWithErrorContent()
+        {
+            bool res = ForgeSchemaValidator.ValidateSchemaAsString(invalidSchemaWithErrorContent, jschemaRules, false, out IList<ValidationError> errorList);
+            Assert.AreEqual(false, res);
+            Assert.AreEqual("JSON is valid against no schemas from 'oneOf'.", errorList.First().Message);
+        }
+
+        [TestMethod]
+        public void Test_ValidateInvalidSchemaAsDictionary()
+        {
+            bool res = ForgeSchemaValidator.ValidateSchemaAsString(invalidSchemaWithErrorContent, jschemaRules, true, out IList<ValidationError> errorList);
+            Assert.AreEqual(false, res);
+            Assert.AreEqual("JSON is valid against no schemas from 'oneOf'.", errorList.First().Message);
+        }
+
+        [TestMethod]
+        public void Test_ValidateValidSchemaAsDictionary()
+        {
+            bool res = ForgeSchemaValidator.ValidateSchemaAsString(subroutineSchemaString, rulesForDictionary, true, out IList<ValidationError> errorList);
+            Assert.AreEqual(true, res);
+            Assert.AreEqual(0, errorList.Count);
+        }
+
+        [TestMethod]
+        public void Test_ValidateAsDictionaryContainsDuplicateKey()
+        {
+            try
+            {
+                ForgeSchemaValidator.ValidateSchemaFromDirectory(schemaDirectoryPath, stringRules, true, out IList<ValidationError> errorList);
+            }
+            catch (Exception e)
+            {
+                Assert.AreEqual("An item with the same key has already been added.", e.Message);
+            }
+        }
+
+        [TestMethod]
+        public void Test_GetLinkedJSchemaRules()
+        {
+            try
+            {
+                JSchema linkedRules = ForgeSchemaValidator.GetLinkedJSchemaRules(stringRules, stringRules, "//ForgeSchemaValidationRules.json");
+                ForgeSchemaValidator.ValidateSchemaAsForgeTree(treeSchema, linkedRules, out IList < ValidationError > errorList);
+            }
+            catch (Exception ex)
+            {
+                Assert.Fail("Expected no exception, but got: " + ex.Message);
+            }      
```

Comment:
Let's rework this UT. Once ForgeSchemaDictionaryValidationRules is updated, any tests with validateAsDictionary as true will be using the linked rules.

If you want to keep a UT specific for GetLinkedJSchemaRules, perhaps we could have an invalid test that links an incorrect referenceUri.